lib/utils.js

```diff
@@ -47,24 +47,30 @@ const unwind = (value, map) => {
 
 const isPromise = (x) => x && typeof x.then === 'function'
 
-const parseErrorEmitEvent = (err, ctx, emitter) => {
+const parseErrorEmitEvent = (result, fullCtx) => {
```

Comment:
Why pass through "this" as `fullCtx`? Why not just keep three args as `(result, ctx, emitter)`?
No need to reassign those vars after.

Reply:
Good catch. There was something there that warranted that push-thru. I'll switch it back.

Reply:
@aptran Oh wait...no, I want it like that. The function signature is smaller now.

Reply:
Hmm.. 2 args vs 3 args isn't that big of a change. I would prefer we keep the args explicit instead of passing through the entire `this`. I guess it's kind of a personal opinion but I would love to hear thoughts from others.
If "it makes the function signature smaller" is the only reason for this change, I'm not 100% convinced it's necessary. It requires the extra steps to assign `ctx` and `emitter`.

Reply:
It’s not 2 vs. 3, it’s 2 vs. 4. Because I’d also have to pass in the `stepName` (i.e. `result.name`).

Reply:
Oh I'm fine with passing through the entire `result`. It's just the `this` part that I don't like.

Reply:
Agree with @aptran here on keeping args explicit wherever possible